resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/internal/AsynchronousInvoke.java

```diff
@@ -24,219 +24,189 @@ public AsynchronousInvoke(ClientInvocation invocation)
    @Override
    public Future<Response> get()
    {
-      invocation.setMethod(HttpMethod.GET);
-      return invocation.submit();
+      return method(HttpMethod.GET);
    }
 
    @Override
    public <T> Future<T> get(Class<T> responseType)
    {
-      invocation.setMethod(HttpMethod.GET);
-      return invocation.submit(responseType);
+      return method(HttpMethod.GET, responseType);
    }
```

Comment:
It's not just simplifications.
The main point is that each time `get`, `trace` etc is called `invocation.setEntity()` (see `method` function) is also called to set the entity to the correct value. It allows to avoid exception such as  `javax.ws.rs.ProcessingException: RESTEASY004565: A GET request cannot have a body.`